src/Adapter/Profile/Permission/QueryHandler/GetPermissionsForConfigurationHandler.php

```diff
@@ -0,0 +1,305 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Profile\Permission\QueryHandler;
+
+use Context;
+use Module;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\Query\GetPermissionsForConfiguration;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\QueryHandler\GetPermissionsForConfigurationHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\QueryResult\ConfigurablePermissions;
+use PrestaShopBundle\Security\Voter\PageVoter;
+use Profile;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Tab;
+
+/**
+ * Get configuratble permissions
+ *
+ * @internal
+ */
+final class GetPermissionsForConfigurationHandler implements GetPermissionsForConfigurationHandlerInterface
+{
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     */
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetPermissionsForConfiguration $query)
+    {
+        $profiles = $this->getProfilesForPermissionsConfiguration();
+        $tabs = $this->getTabsForPermissionsConfiguration();
+        $permissions = ['view', 'add', 'edit', 'delete'];
+
+        $tabPermissionsForProfiles = $this->getTabPermissionsForProfiles($profiles);
+        $modulePermissionsForProfiles = $this->getModulePermissionsForProfiles($profiles);
+
+        $permissionIds = ['view' => 0, 'add' => 1, 'edit' => 2, 'delete' => 3, 'all' => 4];
```

Comment:
```
$permissions = array_keys($permissionIds);
```
and use const for 0, 1, 2, 3, 4.

Reply:
and use Permission::VIEW, Permission::ADD, etc.

Reply:
i know, thats what i was going to do, dont forget its in wip. :wink:

Reply:
@sarjon dont forget it 😄

Reply:
@PierreRambaud Please ;)